jenkins_jobs/modules/parameters.py

```diff
@@ -1143,6 +1143,229 @@ def param_separator(registry, xml_parent, data):
+                        script in a sandbox with limited abilities.
+                    * **script-additional-classpath** (`list`) Additional
+                        classpath entries accessible from the script.
+    :arg bool omit-value-field: By default Dynamic Reference Parameters always
+        include a hidden input for the value. If your script creates an input
+        HTML element, you can check this option and the value input field will
+        be omitted.
+    :arg str referenced-parameters: Comma separated list of other job
+         parameters referenced in the uno-choice script. When any of the
+         referenced parameters are updated, the Groovy script will
+         re-evaluate the choice list using the updated values of referenced
+         parameters.
+    :arg str choice-type: type of the choices. (default 'ET_TEXT_BOX')
+
+        :Allowed Values: * **ET_TEXT_BOX**
+                    * **ET_ORDERED_LIST**
+                    * **ET_UNORDERED_LIST**
+                    * **ET_FORMATTED_HTML**
+                    * **ET_FORMATTED_HIDDEN_HTML**
+    :arg str project-name: the name of job from which the user can pick runs
+    :arg str full-project-name: the full project name of job from which the
+        user can pick runs
+
+    Example::
+
+      parameters:
+        - dynamic-reference:
+            name: lorem
+            description: ipsum
+            script:
+                groovy: |-
+                    return [
+                        'param1',
+                        'param2'
+                    ]
+                use-groovy-sandbox: 'true'
+                script-additional-classpath:
+                    - file:/jar-file-path
+                    - file:/jar-file-path2
+            fallback-script:
+                groovy: |-
+                    return [
+                        'param3',
+                        'param4'
+                    ]
+                use-groovy-sandbox: 'true'
+                script-additional-classpath:
+                    - file:/jar-file-path
+                    - file:/jar-file-path2
+            referenced-parameters: param1, param2
+            choice-type: ET_ORDERED_LIST
+            omit-value-field: False
+            project-name: job-name-test
+            project-full-name: my_jobs/job_folder1/job-name-test
+    """
+    element_name = "org.biouno.unochoice.DynamicReferenceParameter"
+    pdef = XML.SubElement(xml_parent, element_name)
+
+    valid_choice_types = [
+        "ET_TEXT_BOX",
+        "ET_ORDERED_LIST",
+        "ET_UNORDERED_LIST",
+        "ET_FORMATTED_HTML",
+        "ET_FORMATTED_HIDDEN_HTML",
+    ]
+
+    mapping = [
+        ("name", "name", ""),
+        ("description", "description", ""),
+        ("choice-type", "choiceType", "ET_TEXT_BOX", valid_choice_types),
+        ("project-name", "projectName", None),
+        ("project-full-name", "projectFullName", None),
+        ("referenced-parameters", "referencedParameters", ""),
```

Comment:
Does it make sense to have a parameter of this kind without specifying "referenced-parameters" ?

Reply:
This is empty input of the plugin:

![image](https://user-images.githubusercontent.com/4595094/93748795-1f5f7700-fbf9-11ea-9fbe-736e37aca721.png)

What would be your suggestions for it?